plugins/woocommerce-blocks/assets/js/blocks/product-template/edit.tsx

```diff
@@ -107,6 +113,25 @@ const ProductTemplateEdit = ( {
 		12,
 		isNumber
 	);
+
+	const parentCollectionBlockAttributes =
+		useProductCollectionBlockAttributes( clientId );
+
+	const productCollectionBlockContext = useMemo( () => {
```

Comment:
I know this is just a draft and may not be intended to be complete, but extracting this method to a util and adding tests around this would be a nice plus.

Reply:
Good point! I'll go ahead and make those changes.